go-controller/pkg/allocator/ip/subnet/allocator_test.go

```diff
@@ -141,8 +138,18 @@ var _ = ginkgo.Describe("Subnet IP allocator operations", func() {
 			}
 		})
 
+		ginkgo.It("fails to allocate multiple IPs from the same subnet", func() {
+			subnets := []string{"10.1.1.0/24"}
```

Comment:
Let's test it with dual stack to be sure ips at different families don't trigger the guard 
```suggestion
			subnets := []string{"10.1.1.0/24",
				"2000::/64"}
```